File: BehaviorPattern/Memento/Python/Helpers/Memento.py
class Memento:
    def __init__(self, state):
        """
        Initializes the Memento with the given state.
        :param state: The state to save in the Memento.
        """
        self._state = state

    def get_state(self):
        """
        Returns the saved state.
        :return: The state stored in the Memento.
        """
        return self._state

File: BehaviorPattern/Memento/Golang/Helpers/memento.go
package helpers

// Memento stores the state of the Originator
type Memento struct {
	state string
}

// NewMemento creates a new Memento with the given state
func NewMemento(state string) *Memento {
	return &Memento{state: state}
}

// GetState returns the state stored in the Memento
func (m *Memento) GetState() string {
	return m.state
}